From 36df137cc53c7bf652bfb189ab8283691d9692b4 Mon Sep 17 00:00:00 2001
From: Cristy <urban-warrior@imagemagick.org>
Date: Mon, 24 Jul 2017 08:42:22 -0400
Subject: [PATCH] assertion failed in DestroyImage due to mat coder

The upstream patch lead to a compilation error (see #870047).

Thanks to LocutusOfBorg for fixing it.

bug-debian: https://bugs.debian.org/870020, https://bugs.debian.org/870047
bug: https://github.com/ImageMagick/ImageMagick/issues/610,https://github.com/ImageMagick/ImageMagick/pull/627
origin: https://github.com/ImageMagick/ImageMagick/commit/a440f9ea11e0dbefb7a808cbb9441198758b0cb,
 https://github.com/ImageMagick/ImageMagick/pull/627/commits/6df8dd0e8b425b2d7790e1e6aa3237b1c57f05e4
---
 coders/mat.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/coders/mat.c b/coders/mat.c
index cca314144..24e20c11c 100644
--- a/coders/mat.c
+++ b/coders/mat.c
@@ -1356,10 +1356,15 @@ END_OF_READING:
     clone_info = NULL;
   }
   if (logging) (void)LogMagickEvent(CoderEvent,GetMagickModule(),"return");
-  if ((image != image2) && (image2 != (Image *) NULL))
-    image2=DestroyImage(image2);
-  if(image==NULL)
+  if (image==NULL)
+  {
     ThrowReaderException(CorruptImageError,"ImproperImageHeader");
+  }
+  else
+  {
+    if ((image != image2) && (image2 != (Image *) NULL))
+      image2=DestroyImage(image2);
+  }
   return (image);
 }
 
